### ISSUE TITLE
Create test files for additional annotation metadata

### ISSUE THREAD

**Issue by ukemi** [geneontology/gocamgen#77]
Now that the annotations are passing the QC checks (#63), we should start thinking about the additional metadata that MGI would like to import with their annotations. There are six types of metadata that we would like to have imported, probably as comments on the GO-CAM annotations.

1. Created by date
2. modified by date
3. created by person/group
4. modified by person/group
5. any has_participant relations that MGI has as current annotation extensions
6. curator notes

What is the best way to proceed from here? I could make a test file for @dustine32 with just a few annotations. Then I can put a work order in for @loricorbani to generate a GPAD with this information from our database. Of course some of this will also be dependent on how we decide to handle annotation history.

**Comment by goodb** [comment]
It would be useful to make some example go-cam models that have the annotations like you want them.  e.g. are all of these model level annotations or should some of them be added as annotations to node or relations within the models?  We should also try to make use of controlled terminology for annotation properties (e.g. PROV, skos, or our own little ontology) - we don't want everything stuffed into comments if we want to be able to query it later.

**Comment by ukemi** [comment]
@goodb these are all excellent questions that we might want to discuss at one of the Wednesday meetings. In fact, one of the MGI curators was just asking me last Friday where he should be putting the comments in Noctua that he normally attaches to an annotation in the MGI editorial interface. As far as I know, we never made a decision on this. All of these notes are annotation-level notes, but the curator wanted to know if it was ok to just attach them to the main node or if he had to copy them to every piece of evidence etc. I will tag some of his current models because he has already been adding them. We can look at those on a call and decide if we think he is doing it correctly. They are definitely not at the model level. They are usually notes to himself as to why he made the annotation. ping @vanaukenk 
For the has_participant annotation extensions that we are dropping, I would certainly want to be able to query them in the future. I'm not so sure about the curators notes. They are usually used when reviewing annotations.

**Comment by ukemi** [comment]
@loricorbani and I have made the first addition to the MGI gpad file available here:
http://www.informatics.jax.org/downloads/custom/mgi.gpa
This will be the home for our one-time-load **TEST** file for imports.

This GPAD1.0 file contains 'contributor' information when an MGI curator has either created or modified and annotation in MGI. This information is not usually edited by a curator, but is displayed on the Noctua home page in the list of models and shown as un-editable contributor annotations in the model itself. 
Ping @dustine32 @kltm @goodb @vanaukenk 
Once we have tested this addition, @loricorbani and I will continue to add additional metadata properties to our GPAD file. Next we will add created_by and modified_by data, which are also not usually editable by curators.

**Comment by dustine32** [comment]
@ukemi Fantastic! @dougli1sqrd has done a lot of work setting up separate branches of pipeline and the go-site metadata for the import load. Now, with this import GPAD URL, we can start iterating on handling the additional properties and history. And the results will be much easier to manage!

**Comment by dustine32** [comment]
@ukemi @loricorbani Would it be difficult to gzip the file at this location?
http://www.informatics.jax.org/downloads/custom/mgi.gpa

Similar to the regular annotation files deposited to the GO pipeline. Two reasons: faster download (obviously) and also I think the GO pipeline will croak if it's not gzipped. Thanks!

**Comment by kltm** [comment]
@dougli1sqrd Will it still croak? I thought it didn't croak anymore.

**Comment by dustine32** [comment]
Sorry @kltm I think I was referring to when it happened before but didn't realize it was fixed. So, only the one reason for gzipping it (bandwidth) left. Thanks!

**Comment by dougli1sqrd** [comment]
This was an issue in past times. The downloading logic is more robust now and in general will unzip if zipped, and keep uncompressed if found that way. 

(As an aside, this is functionality that also extends to the downloader which we should probably be utilizing, but we are not, as far as the Jenkinsfile goes)

But ontobio handles both at the moment.

**Comment by dustine32** [comment]
Thanks @dougli1sqrd for confirming! No change necessary then @ukemi @loricorbani

**Comment by ukemi** [comment]
@dustine32 , we are going to try to update this file on Fridays. The update file will contain not only additional properties/metadata as we move forward, but the progress that I am making cleaning up the Shex errors in #79. Is there any way that you can give me a status report once in a while as a sanity check that I am making progress?

**Comment by loricorbani** [comment]
I gzipped the file:
http://www.informatics.jax.org/downloads/custom/mgi.gpa.gz
http://www.informatics.jax.org/downloads/custom/mgi.gpa

**Comment by loricorbani** [comment]
new  file 03/27/2020:
http://www.informatics.jax.org/downloads/custom/mgi.gpa.gz
http://www.informatics.jax.org/downloads/custom/mgi.gpa

**Comment by dustine32** [comment]
Thanks @loricorbani ! Finally got to test the contributor parsing and it works great!

Example line from 03/27/2020 import GPAD:
```
MGI     MGI:1920971     part_of GO:0048471      MGI:MGI:3794006|PMID:18163442   ECO:0000314                     20081211        MGI     part_of(EMAPA:17972),part_of(CL:0000018)        contributor=http://orcid.org/0000-0003-3394-9805
```
Translated assertion in this [model](http://68.181.46.18:8910/editor/graph/gomodel:5f9bd4a5-f88d-4cab-b397-bc2fc8f3f4e9):
![image](https://user-images.githubusercontent.com/2678599/78284977-06d99400-74d5-11ea-93a5-f768689f142e.png)

Evidence properties show contributor from line:
![image](https://user-images.githubusercontent.com/2678599/78285474-75b6ed00-74d5-11ea-8a1d-c8ff599763b1.png)

The contributor ORCID even links out to [Dmitry](http://orcid.org/0000-0003-3394-9805):
![image](https://user-images.githubusercontent.com/2678599/78286080-a4cd5e80-74d5-11ea-8e47-d1d671786650.png)

So `contributor` should be good now.

**Comment by ukemi** [comment]
Awesome! I am having trouble getting to your personal server. Do they show up on the menu page too? The next step is for us to add the former MGI curators. @hdrabkin is making sure they are in users.yaml, then I will forward them to @loricorbani and we create a new file. One step at a time....

**Comment by hdrabkin** [comment]
is it OUR Dmitry?

**Comment by ukemi** [comment]
Looks like it is correct. It's definitely not the other one.

**Comment by dustine32** [comment]
@ukemi Sorry about the server issues. Our IT dept has been passing the time lately by reshuffling the network firewall rules on a daily basis. So internet access to this server has been very spotty lately. If you can at least get to the title page, this model's title is "MGI:MGI:1920971-issue-77-contributor".

**Comment by ukemi** [comment]
@dustine32 and @kltm, are we ready now to move on to the created_by, created_date, modified_by, modified_date properties? Do we know where to put them in the models?

**Comment by kltm** [comment]
@ukemi I think there are two things here worth keeping in mind.

First, and most important, is that the created* and modified* fields are not (as far as I know) supported anywhere yet--neither client nor server. If you want to see what these changes will do, we'll have to do some work and then do a mass update of models on `dev` to generally test.

The second thing is relatively minor. At some point we should start moving toward the support of the fields according to the GPAD 2.0 spec. It would be good to have @dustine32 touch bases with @dougli1sqrd and start working towards a timeline for a common codebase.

**Comment by ukemi** [comment]
Hi @kltm. This is exactly why I commented above. As far as I know, these properties will all be retained in GPAD 2.0 in a single field, and they are in the preliminary specs. Adding them now in my mind is still forward progress. As long as the format of the properties is set, I don't think they will hurt anything and adding them will allow us to move forward on our end. It's not completely trivial to request them, write the code and test before they go out the door. As you are aware, the specs for GPAD/GPI are still in flux, so I think it is a bit premature to work on them at our end and then have to redo that work when the specs are finalized. I'd rather add the data to the file that is required and then when the GPAD specs are ready convert our file to 2.0 I hope this makes sense. We can touch base about it at our Thursday meeting. Ping @loricorbani as a heads-up.

**Comment by kltm** [comment]
@ukemi Agreed for touching bases on this again and agreed that getting holding data in place is likely an overall good. For the second point, I don't think these properties are actually on spec for 2.0 (https://github.com/geneontology/go-annotation/blob/master/specs/gpad-gpi-2-0.md#gpad-annotation-properties-proposed)--there is a difference between "_" and "-". That said, as the created\* and modified\* are not being processed at this point, it's not critical (but would be preferable as it is the test data at this point).

**Comment by ukemi** [comment]
LOL. I knew you would get me on the - versus _. We will do it according to the spec.  :)

**Comment by dustine32** [comment]
@ukemi Yeah, if you guys want to start including the created*, modified* fields in the test import file, I can at least write a unit test to ensure they're being parsed and populated into the python `Evidence` class objects correctly. That'll be _some_ progress!

**Comment by loricorbani** [comment]
new file 04/16/2020:
http://www.informatics.jax.org/downloads/custom/mgi.gpa.gz
http://www.informatics.jax.org/downloads/custom/mgi.gpa

**Comment by ukemi** [comment]
@dustine32, this file includes the contributor information for all MGI curators, current and former. It will also be a test file for the contributors who don't have orcids and are listed as GOC:$$$.

**Comment by dustine32** [comment]
@ukemi @loricorbani Finally got around to testing the `GOC:$$$` contributor values in the new test import GPAD (dated 2020-04-29). So far only checked one model (for [MGI:MGI:108450](http://68.181.46.18:8910/editor/graph/gomodel:9e86e4ee-fce6-46bd-b633-e4046e970989)) containing `contributor=GOC:apd` from this line:
```
MGI     MGI:108450      enables GO:0004016      MGI:MGI:80863|PMID:8662814      ECO:0000314                     20160525        MGI             contributor=GOC:apd|contributor=http://orcid.org/0000-0001-7476-6306
```
![image](https://user-images.githubusercontent.com/2678599/80631086-15e72f80-8a0a-11ea-9320-d3d105870d41.png)

![image](https://user-images.githubusercontent.com/2678599/80630800-a07b5f00-8a09-11ea-93b4-5b7d3f0c4f58.png)
I think this is as good as we expected for translation. What do you think?

**Comment by ukemi** [comment]
Hi @dustine32. Yes. For the former curators who don't have orcIDs, this is the best we can do. These are all valid users in the users.yaml file, so I think this is what we would expect. @kltm do you see any issues?

**Comment by ukemi** [comment]
We should have a closer look at where these data are being added. They are being added to evidence, but there are annotations associated with individuals that might be more appropriate. Let's look one on of the GO-CAM specs calls. @vanaukenk

**Comment by goodb** [comment]
@ukemi I think this should really be an issue in the go_shapes repo as it impacts all go_cam models and associated software.

**Comment by ukemi** [comment]
You are correct. I think @vanaukenk and I need to have a look and make a proposal for a ticket.

**Comment by dustine32** [comment]
Linking the properties `comment` field ticket #87 here.
